tests/server/test_compilerservice.py

```diff
@@ -2016,3 +2023,84 @@ async def test_overlapping_env_vars(mocked_compiler_service, server, client, env
             env_vars={"var": "val", "test": "123"},
             mergeable_env_vars={"var": "otherval", "somekey": "someval"},
         )
+
+
+class Dudreport:
+
+    async def update_streams(self, out: str = "", err: str = "") -> None:
+        pass
+
+
+async def test_venv_use_and_reuse(tmp_path, caplog):
+    caplog.at_level(logging.DEBUG)
+    project = tmp_path / "project"
+
+    venv = project / ".env"
+
+    project.mkdir()
+    run = CompileRun(None, str(project))
+    # Move for logging
+    run.stage = Dudreport()
+
+    # Create
+    await run.ensure_compiler_venv()
+    assert venv.exists()
+    log_contains(caplog, "inmanta.server.services.compilerservice", logging.INFO, "Creating new venv at")
+    caplog.clear()
+
+    # re-use
+    await run.ensure_compiler_venv()
+    log_contains(caplog, "inmanta.server.services.compilerservice", logging.INFO, "Found existing venv")
+    caplog.clear()
+
+
+async def test_venv_upgrade_capture(tmp_path, caplog):
```

Comment:
I don't really understand what the word capture mean in this name and that of the next test case

Reply:
improved, thx